tests/test/test_translation.py

```diff
@@ -0,0 +1,157 @@
+import json
+
+from bs4 import BeautifulSoup as bs4
+from django.test import override_settings
+from django.urls import reverse
+from django.utils import translation
+from wagtail.models import Locale, Page
+from wagtail.test.utils import TestCase, WagtailTestUtils
+
+from wagtail_footnotes.models import Footnote
+
+from ..models import TestPageStreamField
+
+
+@override_settings(
+    LANGUAGES=[
+        ("en", "English"),
+        ("fr", "French"),
+        ("de", "German"),
+    ],
+    WAGTAIL_CONTENT_LANGUAGES=[
+        ("en", "English"),
+        ("fr", "French"),
+        ("de", "German"),
+    ],
+)
+class TestSubmitPageTranslationView(WagtailTestUtils, TestCase):
+    def setUp(self):
+        self.en_locale = Locale.objects.first()
+        self.fr_locale = Locale.objects.create(language_code="fr")
+        self.de_locale = Locale.objects.create(language_code="de")
+
+        self.en_homepage = Page.objects.get(title="Welcome to your new Wagtail site!")
+        self.fr_homepage = self.en_homepage.copy_for_translation(self.fr_locale)
+        self.fr_homepage.save_revision().publish()
+        self.de_homepage = self.en_homepage.copy_for_translation(self.de_locale)
+        self.de_homepage.save_revision().publish()
+
+        self.uuid = "f291a4b7-5ac5-4030-b341-b1993efb2ad2"
+        self.en_test_page = TestPageStreamField(
+            title="Test Page With Footnote",
+            slug="test-page-with-footnote",
+            body=json.dumps(
+                [
+                    {
+                        "type": "paragraph",
+                        "value": f'<p>This is a paragraph with a footnote. <footnote id="{self.uuid}">1</footnote></p>',
+                    },
+                ]
+            ),
+        )
+        self.en_homepage.add_child(instance=self.en_test_page)
+        self.en_test_page.save_revision().publish()
+        self.en_footnote = Footnote.objects.create(
+            page=self.en_test_page,
+            uuid=self.uuid,
+            text="This is a footnote",
+        )
+
+    def test_translating_page_translates_footnote(self):
+        url = reverse(
+            "simple_translation:submit_page_translation", args=(self.en_test_page.id,)
+        )
+        self.login()
```

Comment:
given you run `self.login()` in each test, move it to `setUp()`.
The url can be moved to `setUpTestData` since you're using it more than once

Reply:
Done